### PR TITLE
Extension registration

### LINKED ISSUE #2595: Allow extensions to dynamically register roles
Plugins can register roles by adding new roles to [roles.yml](https://github.com/opensearch-project/security/blob/main/config/roles.yml). The roles are added to the cluster on bootstrap of a fresh cluster installation, by running securityadmin.sh or by using the API to create the roles. 

Extensions need a way to register roles with a list of actions and have the roles sourced to the opensearch security index on extension installation.

### PR DESCRIPTION
**Pull request by samuelcostae**
### Description
Draft PR to get feedback and questions .
This changes aim to Provide an endpoint where Extension can register themselves in the cluster and list its protected indices that should not be accessed  by any user.

 Category: new feature
  

### Issues Resolved
#2595

### Testing
A unique unit  test added so far to assist devlopment 

### Check List
- [ ] New functionality includes testing
- [ ] New functionality has been documented
- [x ] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by peternied — CHANGES_REQUESTED** [review]
It looks like this is duplicating the list of extensions that are specified by the OpenSearch extension yaml file and then loaded into the extension manager.

What is you're thinking about reconcile having two sources of information about extensions?

**Comment by samuelcostae** [comment]
@peternied Thanks a million for the offer, I eventually figured out the put  the request was arriving at the new class, and it was because i hadn't implemented a Validator (Added a dummy one in the meantime to always return true).

I have two questions here regarding the requests.  I'm not sure what is the process to  add a document to a indice in the OS  cluster. In the draft PR i have i used  
`        client.index(new IndexRequest("index_name").source(contentAsNode)); `

Is that the correct approach to add a document in a cluster?

I also think im missing something pretty obvious regarding why the request's body im sending here on ExtensionRegistrationApiActionTest.java 

`        RestHelper.HttpResponse responsea = keyStoreRestHelper.executePutRequest( ENDPOINT, correctExtensionRequest);`

AS the body / content is arriving empty, even though from other tests it seems pretty straight forward.

Thanks

**Comment by samuelcostae** [comment]
> It looks like this is duplicating the list of extensions that are specified by the OpenSearch extension yaml file and then loaded into the extension manager.
> 
> What is you're thinking about reconcile having two sources of information about extensions?



From the discussion on #2553,  I had understood that  extensions would communicate with OS or Security via REST  requests, mainly the posts before/after  this @cwperks 's post:    https://github.com/opensearch-project/security/issues/2553#issuecomment-1526825194 . 

> What is you're thinking about reconcile having two sources of information about extensions?

Having two sources is most of times not good, on this PR I had no intention to add a new source of truth  for the extension data. My understanding is that wherever the extension would have its information stored (either the settings.yaml or whichever external souce to be defined/discussed in the future), this endpoint would  be touched with that info and the relevant info would be  saved to the index.

But I might be missing some context regarding how the best approach to saving info to internally an index.

File: src/main/java/org/opensearch/security/dlic/rest/api/ExtensionRegistrationApiAction.java [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+            throws IOException{
+
+        createdResponse(channel, " updated");
+
+        ;
+    }
+
+    @Override
+    protected void handlePut(RestChannel channel, final RestRequest request, final Client client, final JsonNode content) throws IOException {
+
```
Comment (samuelcostae) [inline src/main/java/org/opensearch/security/dlic/rest/api/ExtensionRegistrationApiAction.java:104]: The request body  arrives here empty .

An im missing something obvious?

**Pull request closed without merging**